src/analytics.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * Analytics for server-side events.
+ *
+ * @module analytics
+ */
+import { default as UserAgentBag } from "user-agent-bag";
+import { PLAUSIBLE_ANALYTICS_DOMAIN, PLAUSIBLE_ANALYTICS_ENDPOINT } from "./config.js";
+import { fetch } from "./fetch.js";
+
+
+/**
+ * Generate middleware to record an analytics event via a request to
+ * Plausible's API.
+ *
+ * Intended for use in an Express routing chain.  The generated middleware
+ * function does not block on the analytics request before returning.
+ *
+ * If the User-Agent of the incoming request starts with "Nextstrain-CLI/",
+ * then it is parsed for several bits of information and those are
+ * automatically attached to the analytics event as custom properties.
+ *
+ * @function recordEvent
+ * @param {object} data
+ * @param {string} data.name - Event name.  Defaults to "pageview" (Plausible's default).
+ * @param {object} data.props - Custom properties.  Defaults to {}.
+ * @param {boolean} data.interactive - Interactive session or not?  Defaults to
+ *   false (opposite of Plausible's default), with the reasoning that most events
+ *   we're likely to record server-side aren't for interactive browser-based
+ *   sessions.
+ * @returns {expressMiddlewareAsync}
+ */
+export const recordEvent = ({name = "pageview", props = {}, interactive = false} = {}) => async (req, res, next) => {
+  if (PLAUSIBLE_ANALYTICS_DOMAIN) {
+    /* We intentionally do not "await fetch()" as we do not want to block
+     * request processing on analytics; we'd rather ignore failures.
+     *    -trs, 28 April 2025
+     */
+    fetch(
+      // <https://plausible.io/docs/events-api>
+      PLAUSIBLE_ANALYTICS_ENDPOINT, {
```

Comment:
I'm surprised this doesn't require some king of API key, but maybe that's normal for tracking?

Reply:
It's pretty normal for analytics that are focused on the client-side (like Plausible), since there is no way to have any actual secrets without server-side involvement. Analytics spam is totally a thing that happens as a result, though, and as such providers like Plausible have to do a bit of spam/abuse prevention work.